cms/src/db/repositories/baseRepository.ts

```diff
@@ -19,6 +19,44 @@ export class BaseRepository {
         return db.docs.where("parentId").equals(parentId);
     }
 
+    whereParentIds(parentIds: Uuid[]) {
+        return db.docs.where("parentId").anyOf(parentIds);
+    }
+
+    whereLanguageIds(languageIds: Uuid[]) {
+        return db.docs.where("language").anyOf(languageIds);
+    }
+
+    /**
+     * Return a list of documents and change documents of specified DocType that are NOT members of the given groupIds
+     */
+    whereNotMemberOf(groupIds: Array<Uuid>, docType: DocType) {
+        // Query groups and group changeDocs
+        if (docType === DocType.Group) {
+            return db.docs.filter(
+                (group) =>
+                    (group.type === DocType.Group ||
+                        (group.type === DocType.Change && group.docType === DocType.Group)) &&
+                    group.acl! &&
```

Comment:
Is this `!` needed here? Shouldn't matter too much but if it's not it feels a bit cleaner.

Reply:
TypeScript checks are failing if I remove the "!". Guess it is one of the joys of using TypeScript...